pkg/dpm/plugin.go

```diff
@@ -74,17 +74,19 @@ func (dpi *devicePlugin) StartServer() error {
 		return nil
 	}
 
-	if err := dpi.register(); err != nil {
-		glog.Errorf("error registering with device plugin manager: %v", err)
-		return err
-	}
 	err := dpi.serve()
 	if err != nil {
 		return err
 	}
 
 	dpi.Running = true
```

Comment:
I know this isn't a new bug, but while we have the patient open...should this be below the invocation of `register()`?  I think if register fails we do want to retry later.

Reply:
That could be the fix to the original issue that motivated the PR being reverted. Good catch.

Will push it.

Reply:
Done.